src/libraries/System.Security.Cryptography/tests/Shake128Tests.cs

```diff
@@ -0,0 +1,1557 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Security.Cryptography.Tests
+{
+    public class Shake128Tests : ShakeTestDriver<Shake128Tests.Traits, Shake128>
+    {
+        public class Traits : IShakeTrait<Shake128>
+        {
+            public static Shake128 Create() => new Shake128();
+            public static bool IsSupported => Shake128.IsSupported;
+
+            public static void AppendData(Shake128 shake, byte[] data) => shake.AppendData(data);
+            public static void AppendData(Shake128 shake, ReadOnlySpan<byte> data) => shake.AppendData(data);
+            public static byte[] GetHashAndReset(Shake128 shake, int outputLength) => shake.GetHashAndReset(outputLength);
+            public static void GetHashAndReset(Shake128 shake, Span<byte> destination) => shake.GetHashAndReset(destination);
+            public static byte[] GetCurrentHash(Shake128 shake, int outputLength) => shake.GetCurrentHash(outputLength);
+            public static void GetCurrentHash(Shake128 shake, Span<byte> destination) => shake.GetCurrentHash(destination);
+
+            public static byte[] HashData(byte[] source, int outputLength) => Shake128.HashData(source, outputLength);
+            public static byte[] HashData(ReadOnlySpan<byte> source, int outputLength) => Shake128.HashData(source, outputLength);
+            public static void HashData(ReadOnlySpan<byte> source, Span<byte> destination) => Shake128.HashData(source, destination);
+
+            public static byte[] HashData(Stream source, int outputLength) => Shake128.HashData(source, outputLength);
+            public static void HashData(Stream source, Span<byte> destination) => Shake128.HashData(source, destination);
+
+            public static ValueTask HashDataAsync(Stream source, Memory<byte> destination, CancellationToken cancellationToken) =>
+                Shake128.HashDataAsync(source, destination, cancellationToken);
+
+            public static ValueTask<byte[]> HashDataAsync(Stream source, int outputLength, CancellationToken cancellationToken) =>
+                Shake128.HashDataAsync(source, outputLength, cancellationToken);
+
+            public static ValueTask HashDataAsync(Stream source, Memory<byte> destination) =>
+                Shake128.HashDataAsync(source, destination);
+
+            public static ValueTask<byte[]> HashDataAsync(Stream source, int outputLength) =>
+                Shake128.HashDataAsync(source, outputLength);
+        }
+
+        protected override IEnumerable<(string Msg, string Output)> Fips202Kats
+        {
+            get
+            {
+                // SHAKE128VariableOut.rsp
```

Comment:
this this test set coming from the RFC?

Reply:
They come from NIST / CAVP. They are in the "SHA-3 XOF Test Vectors for Byte-Oriented Output" file from https://csrc.nist.gov/Projects/cryptographic-algorithm-validation-program/Secure-Hashing.